test/test_filesystem_helper.cpp

```diff
@@ -103,3 +103,53 @@ TEST(TestFilesystemHelper, is_absolute)
     }
   }
 }
+
+TEST(TestFilesystemHelper, correct_extension)
+{
+  if (is_win32) {
+    {
+      auto p = path("C:\\foo\\bar\\baz.yml");
+      auto ext = p.extension();
+      EXPECT_EQ(".yml", ext);
+    }
+    {
+      auto p = path("C:\\foo\\baz.bar.yml");
+      auto ext = p.extension();
+      EXPECT_EQ(".yml", ext);
+    }
+  } else {
+    {
+      auto p = path("/foo/bar/baz.yml");
+      auto ext = p.extension();
+      EXPECT_EQ(".yml", ext);
+    }
+    {
+      auto p = path("/foo/baz.bar.yml");
+      auto ext = p.extension();
+      EXPECT_EQ(".yml", ext);
+    }
+  }
+}
+
+TEST(TestFilesystemHelper, is_empty)
+{
+  {
+    auto p = path("");
+    EXPECT_TRUE(p.empty());
+  }
+}
+
+TEST(TestFilesystemHelper, create_directories)
```

Comment:
This should probably have a comment stating that this expected the current directory to be write-only (otherwise the test would fail). This is not great but I don't have an easy way to improve this.